datafusion/substrait/tests/cases/function_test.rs

```diff
@@ -19,40 +19,33 @@
 
 #[cfg(test)]
 mod tests {
+    use crate::utils::test::TestSchemaCollector;
     use datafusion::common::Result;
-    use datafusion::prelude::{CsvReadOptions, SessionContext};
     use datafusion_substrait::logical_plan::consumer::from_substrait_plan;
     use std::fs::File;
     use std::io::BufReader;
     use substrait::proto::Plan;
 
     #[tokio::test]
     async fn contains_function_test() -> Result<()> {
-        let ctx = create_context().await?;
-
         let path = "tests/testdata/contains_plan.substrait.json";
-        let proto = serde_json::from_reader::<_, Plan>(BufReader::new(
+
+        let proto_plan = serde_json::from_reader::<_, Plan>(BufReader::new(
             File::open(path).expect("file not found"),
         ))
         .expect("failed to parse json");
 
-        let plan = from_substrait_plan(&ctx, &proto).await?;
+        let ctx = TestSchemaCollector::generate_context_from_plan(&proto_plan);
+        let plan = from_substrait_plan(&ctx, &proto_plan).await?;
 
         let plan_str = format!("{}", plan);
 
         assert_eq!(
             plan_str,
-            "Projection: nation.b AS n_name\
-            \n  Filter: contains(nation.b, Utf8(\"IA\"))\
-            \n    TableScan: nation projection=[a, b, c, d, e, f]"
+            "Projection: nation.n_name\
+            \n  Filter: contains(nation.n_name, Utf8(\"IA\"))\
+            \n    TableScan: nation projection=[n_nationkey, n_name, n_regionkey, n_comment]"
```

Comment:
You can see in this change how the DataFusion and Substrait plans had different schemas.